### PR TITLE
AsyncSceneInheritOutline to use SceneInstanceReady / be reversible

### PR DESCRIPTION
**Pull request by komadori**
This attempts to address:
* @Shatur's request in #30 that that `AsyncSceneInheritOutline` should use the `SceneInstanceReady` event
* @rparrett's request in #34 that removing `AsyncScenenInheritOutline` should reverse its effect (and by implication that it can be added after the scene has loaded).

### CODE REVIEW

File: src/scene.rs [hunk 1]
```diff
@@ -1,25 +1,95 @@
...
+    >,
+) {
+    for (entity, instance) in async_query.iter() {
+        instance_maybe_ready(commands.reborrow(), &scene_spawner, entity, instance);
+    }
+}
+
+// Handles scenes which become ready after `AsyncSceneInheritOutline` is added.
+#[allow(clippy::type_complexity)]
+fn async_ready_event(
```
Comment (Shatur) [inline src/scene.rs:54]: Why do you need both systems? Isn't `SceneInstanceReady` emitted every time after an instance spawn?

Comment (komadori) [reply src/scene.rs:54]: This supports the use case that you first spawn a scene, the scene becomes ready, and then only later add the outlining components to it.

Personally, I always add the outline components at spawning time and never remove them, but I can't say that an approach where you add and remove the components is wrong. I don't know how efficient the `Added` query filter is though, so perhaps this is no better than not using the event.

Comment (Shatur) [reply src/scene.rs:54]: Makes sense.

`Added` query is a syntax sugar for iterating and checking `is_added()` on `Ref` or `&mut` components.

**Comment by komadori** [comment]
This PR is superseded by commit b3446c5e6d660ca476224904d8909527a1c3abc7, which uses component hooks and observers to efficiently handle adding and removing `AsyncSceneInheritOutline`.

**Pull request closed without merging**